packages/runtime/container-runtime/src/containerRuntime.ts

```diff
@@ -2995,10 +3032,19 @@ export class ContainerRuntime extends TypedEventEmitter<IContainerRuntimeEvents>
                 totalChunks: chunkN,
             };
             offset += maxOpSize;
+
+            let serializedChunkedOp: string | undefined;
```

Comment:
Chunked messages are disabled right now (see this._maxOpSizeInBytes check and defaultMaxOpSizeInBytes)
When/if we enable them, we should do compression differently - we should compress first, then chunk, not other way around.
I'd suggest to simply add assert(false) at the top of this function (with comment what someone needs to do to enable it back) and comment out whole function (no need to may bundle cost for nothing). You can add the code you are adding, but it's wrong in direction, as compression ratio for chunks will be much lower if we compressed whole content first.
That all said, there is no harm in adding your code as is (as it should not run). Your choice.

If you want to - you can enable it properly, this requires move code you are adding in submitRuntimeMessage() to top of submitMaybeChunkedMessages(). But it is not simple - not clear how compression property should apply (to only last chunk? all chunks?) and it makes it harder to deal with op processing, so probably not worth attempting in this PR.

Reply:
I'm going to submit this as-is for now. I've created [this task](https://dev.azure.com/fluidframework/internal/_workitems/edit/1512) for following up and doing chunking correctly.